docs/standards/smart-contracts/utils/lsp2-json-schema-utils.md

```diff
@@ -0,0 +1,347 @@
+---
+title: LSP2JSONSchemaUtils
+sidebar_position: 1
+---
+
+# LSP2JSONSchemaUtils
+
+:::info
+
+[`LSP2Utils.sol`](https://github.com/lukso-network/lsp-smart-contracts/blob/develop/contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol)
+
+:::
+
+This library can be used to generate ERC725Y keys according to the JSON schemas defined in the LSP2 standard.
+
+## Functions
+
+### generateSingletonKey
+
+```solidity
+function generateSingletonKey(
+    string memory keyName
+) internal pure returns (bytes32);
+```
+
+Generates a data key of `keyType` Singleton.
+
+#### Parameters:
+
+| Name      | Type   | Description                                          |
+| :-------- | :----- | :--------------------------------------------------- |
+| `keyName` | string | The string to hash to generate a Singleton data key. |
+
+#### Return Values:
+
+| Name     | Type    | Description                      |
+| :------- | :------ | :------------------------------- |
+| `result` | bytes32 | Data key of `keyType` Singleton. |
+
+### generateArrayKey
+
+```solidity
+function generateArrayKey(
+    string memory keyName
+) internal pure returns (bytes32);
+```
+
+Generates a data key of `keyType` Array by hashing `keyName`.
+
+##### Requirements:
+
+- The last two characters should be `[]`.
+
+E.g.:
+
+```solidity
+string memory keyName = "ArrayName[]";
+```
+
+#### Parameters:
+
+| Name      | Type   | Description                                                            |
+| :-------- | :----- | :--------------------------------------------------------------------- |
+| `keyName` | string | The string that will be used to generate a data key of `keyType` Array |
+
+#### Return Values:
+
+| Name     | Type    | Description                  |
+| :------- | :------ | :--------------------------- |
+| `result` | bytes32 | Data key of `keyType` Array. |
+
+### generateArrayElementKeyAtIndex
+
+```solidity
+function generateArrayElementKeyAtIndex(
+    bytes32 arrayKey,
+    uint256 index
+) internal pure returns (bytes32);
+```
+
+Generates an Array index data key by concatenating the first 16 bytes of `arrayKey` with an `index`.
```

Comment:
Generates the element key of the array at the given index , something like this ..